src/commands/api/findTreeItem.ts

```diff
@@ -26,14 +27,29 @@ export async function findTreeItem(query: TreeItemQuery): Promise<DatabaseAccoun
         context.errorHandling.suppressDisplay = true;
         context.errorHandling.rethrow = true;
 
-        const connectionString = query.connectionString;
         let parsedCS: ParsedConnectionString;
-        if (/^mongodb[^:]*:\/\//i.test(connectionString)) {
-            parsedCS = await parseMongoConnectionString(connectionString);
-        } else if (/^postgres:\/\//i.test(connectionString)) {
+        if (query.postgresData) {
+            const postgresData = query.postgresData;
+            let connectionString: string = `postgres://`;
+            if (postgresData.username && postgresData.password) {
+                const encodedUsername = encodeURIComponent(postgresData.username);
+                const encodedPassword = encodeURIComponent(postgresData.password);
+                connectionString += `${encodedUsername}:${encodedPassword}@`;
+
+            }
+            connectionString += `${postgresData.hostName}:${postgresData.port}`;
+            if (postgresData.dbName) {
+                const encodeDatabaseName = encodeURIComponent(postgresData.dbName);
+                connectionString += `/${encodeDatabaseName}`;
+            }
             parsedCS = parsePostgresConnectionString(connectionString);
         } else {
-            parsedCS = parseDocDBConnectionString(connectionString);
+            const connectionString = nonNullProp(query, 'connectionString');
+            if (/^mongodb[^:]*:\/\//i.test(connectionString)) {
+                parsedCS = await parseMongoConnectionString(connectionString);
+            } else {
```

Comment:
I know this won't be used, but I feel like it's safest to keep this code in here:
```
} else if (/^postgres:\/\//i.test(connectionString)) {
    parsedCS = parsePostgresConnectionString(connectionString);
}
```

Basically, people who want to find a postgres server can search by connection string _or_ postgresData